src/hooks/users.ts

```diff
@@ -1,52 +1,61 @@
 import { useMemo } from "react";
-import { useFirestoreConnect } from "react-redux-firebase";
+import { isLoaded } from "react-redux-firebase";
 
 import { usersSelector, usersByIdSelector } from "utils/selectors";
 import { WithId } from "utils/id";
 
 import { useSelector } from "hooks/useSelector";
 import { useUserLastSeenThreshold } from "hooks/useUserLastSeenThreshold";
 import { useVenueId } from "hooks/useVenueId";
+import { useSparkleFirestoreConnect } from "hooks/useSparkleFirestoreConnect";
 
 import { User } from "types/User";
 
 const useConnectUsers = () => {
-  useFirestoreConnect("users");
+  const venueId = useVenueId();
+
+  useSparkleFirestoreConnect([
+    {
+      collection: "users",
+      where: ["enteredVenueIds", "array-contains", venueId],
+      storeAs: "users",
+    },
+  ]);
 };
 
-export const usePartygoers = () => {
+export const usePartygoers = (): readonly WithId<User>[] => {
   useConnectUsers();
 
   const lastSeenThreshold = useUserLastSeenThreshold();
 
-  const users = useSelector(usersSelector);
+  const users = useSelector(usersSelector) ?? [];
 
   return useMemo(
     () => users.filter((user) => user.lastSeenAt > lastSeenThreshold),
     [users, lastSeenThreshold]
   );
 };
 
-export const useUsersById = () => {
+export const useUsersIsLoaded = () => {
   useConnectUsers();
 
-  return useSelector(usersByIdSelector);
+  const users = useSelector(usersSelector);
+
+  return useMemo(() => isLoaded(users), [users]);
 };
```

Comment:
I think it doesn't make sense for us to have `useConnectUsers` in this hook, as that would sort of impact/change whether the data is loaded.

It should be enough to just have the `useSelector` + `isLoaded` I think.

Also not sure if there is any point in the `useMemo` here, I expect `isLoaded` is probably quite a 'lightweight' function call, so we probably hurt our performance more by memo'ing it than we would gain.

[nit] If I was naming this I would probably call it `useIsUsersLoaded` to follow our boolean naming patterns.

Reply:
Not sure if I should remove `useConnectUsers` from this hook. It will affect the legacy logic, I think.